### PR TITLE
Superbeeny/adding podspec secrets

### PR DESCRIPTION
**Pull request by superbeeny**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [X] [Read the contribution guide](https://docs.radapp.io/community/contributing/docs/)
- [X] Commands include options for Linux, MacOS, and Windows within codetabs
- [X] New file and folder names are globally unique
- [X] Page references use shortcodes instead of markdown or URL links
- [X] Images use HTML style and have alternative text
- [X] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Adding example for the how-to's showing how to use podspec to inject secrets as environment variables into the container.

## Issue reference
Based on conversation in [discord](https://discord.com/channels/1113519723347456110/1222556603153383554)

### CODE REVIEW

**Comment by superbeeny** [comment]
Fixes #1101

**Comment by willtsai** [comment]
Thanks @superbeeny! I will give this a review here within the next day.

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+- Patch existing Kubernetes secrets using [PodSpec](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec) definitions and provide them to the environment of a container.
+
+## Prerequisites
+
+- [rad CLI]({{< ref getting-started >}})
+- [Radius initialized with `rad init`]({{< ref howto-environment >}})
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+## Step 1: Define a container
+Begin by creating a file named `app.bicep` with a Radius [container]({{< ref "guides/author-apps/containers" >}}):
```
Comment (willtsai) [inline docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md]:
```suggestion
## Step 1: Define a container

Begin by creating a file named `app.bicep` with a Radius [container]({{< ref "guides/author-apps/containers" >}}):
```

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+{{< rad file="snippets/secrets-container.bicep" embed=true >}}
+
+## Step 2: Deploy the app and container
```
Comment (willtsai) [inline docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md:28]:
```suggestion
## Step 2: Deploy the app and container

Run this command to deploy the app and container:

```

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+dashboard-7f7db87c5-7d2jf dashboard [port-forward] connected from localhost:7007 -> ::7007
+demo-7d94db59f6-ps6cf demo [port-forward] connected from localhost:3000 -> ::3000
+```
+
+Verify the pod is running:
+    
+```bash
+kubectl get pods -n dev-demo
+```
+
```
Comment (willtsai) [inline docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md:73]: Can you add a code block here to show an example of what the console output would look like? For example:

> You should see the following output in your console: 
> ```
> output
> ```

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+{{< rad file="snippets/secrets-container.bicep" embed=true >}}
+
+## Step 2: Deploy the app and container
+```bash
+rad run ./app.bicep -a demo
+```
+
+Once the deployment completes successfully, you should see the following confirmation message along with some system logs:
+
+```bash
```
Comment (willtsai) [inline docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md]: for depicting console output, you can remove the `bash` tag in the code block

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+kubectl get secrets -n dev-demo
+```
+
+## Step 4: Patch the secret
+
+Patch the secret into the container by adding the following `runtimes` block to the `container` resource in your `app.bicep` file:
+
+{{< rad file="snippets/secrets-patch.bicep" embed=true >}}
```
Comment (willtsai) [inline docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md]: I suggest adding a config here that highlights the lines that are added to the `app.bicep`

```suggestion
{{< rad file="snippets/secrets-patch.bicep" embed=true markdownConfig="{linenos=table,hl_lines=[\"25-60\"]}" >}}
```

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+```bash
+rad app deploy demo
+```
+
+Once the deployment completes successfully, you should see the environment variable in the container:
+First, get the pod name:
+```bash
+kubectl get pods -n dev-demo
+```
```
Comment (willtsai) [inline docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md]:
```suggestion
Once the deployment completes successfully, you should see the environment variable in the container.

To validate this, first get the pod name:

```bash
kubectl get pods -n dev-demo
```
```

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
```
Comment (willtsai) [inline docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md:109]: Can you add a code block to show the output containing the pod name as a result of the above command?

**Review by willtsai — CHANGES_REQUESTED** [review]
This is fantastic! Thank you for adding this How-To Guide. I've left some minor suggestions, please take a look and let me know if you have any questions.

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md (hunk 3 above)
Comment (superbeeny) [reply docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md:73]: Updated to incorporate

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md (hunk 4 above)
Comment (superbeeny) [reply docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md]: Updated to incorporate

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md (hunk 7 above)
Comment (superbeeny) [reply docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md:109]: Updated to incorporate

File: docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md [hunk 8]
```diff
@@ -61,7 +65,11 @@ Verify the pod is running:
 ```bash
 kubectl get pods -n dev-demo
 ```
-
+You should see the following output in your console:
+```
+NAME                    READY   STATUS    RESTARTS   AGE
+demo-7d94db59f6-k7dfb   1/1     Running   0          62s
+```
 ## Step 3: Create a secret
```
Comment (willtsai) [inline docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md:73]:
```suggestion
```

## Step 3: Create a secret
```

Comment (willtsai) [inline docs/content/guides/author-apps/kubernetes/how-to-access-secrets/index.md:74]: committing a minor nit to add a new line here

```suggestion

## Step 3: Create a secret
```

**Review by willtsai — APPROVED** [review]
looks good, thanks for addressing the feedback!

**Pull request merged**